internal/provider/resource_port_test.go

```diff
@@ -2,65 +2,57 @@ package provider
 
 import (
 	"fmt"
+	"os"
 	"testing"
 
 	"github.com/PacketFabric/terraform-provider-packetfabric/internal/testutil"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func hclPort(description, accountUUID, speed, media, subscriptionTerm, pop, zone string, autoneg, nni bool) (hcl string, resourceName string) {
+func hclPort(description, media, speed, pop, subscriptionTerm string) (hcl string, resourceName string) {
 	hclName := testutil.GenerateUniqueResourceName()
 	resourceName = "packetfabric_port." + hclName
-	hcl = fmt.Sprintf(`
-	resource "packetfabric_port" "%s" {
-		description       = "%s"
-		account_uuid      = "%s"
-		speed             = "%s"
-		media             = "%s"
-		subscription_term = "%s"
-		pop               = "%s"
-		zone              = "%s"
-		autoneg           = %t
-		nni               = %t
-	}`, hclName, description, accountUUID, speed, media, subscriptionTerm, pop, zone, autoneg, nni)
+	hcl = fmt.Sprintf(testutil.RResourcePort, hclName, description, media, pop, speed, subscriptionTerm)
 	return
 }
 
 func TestAccPort(t *testing.T) {
 	testutil.SkipIfEnvNotSet(t)
 
-	description := testutil.GenerateUniqueName(testPrefix)
-	pop, zone, err := testutil.GetPopAndZoneWithAvailablePort("1Gbps", "LX")
-	if err != nil {
-		t.Fatalf("Unable to find pop and zone with available port: %s", err)
-	}
+	description := os.Getenv(testutil.PF_PORT_DESCR_KEY)
+	media := os.Getenv(testutil.PF_PORT_MEDIA_KEY)
```

Comment:
We could determine media by speed and pop. I don't see any reason to provide it externally.

Reply:
Is there any API that returns it?

Reply:
If you look in the provider, the location resources should have it. The following data-source returns the speed of a pop:

https://registry.terraform.io/providers/PacketFabric/packetfabric/latest/docs/data-sources/packetfabric_locations_port_availability

And this one the pop available

https://registry.terraform.io/providers/PacketFabric/packetfabric/latest/docs/data-sources/packetfabric_locations

Reply:
We added a logic to get the next available `pop, zone and media` for Dev env.